src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1833,7 +1833,7 @@ bool LibraryCallKit::inline_vector_convert() {
     } else if (num_elem_from > num_elem_to) {
       // Since number elements from input is larger than output, simply reduce size of input (we are supposed to
       // drop top elements anyway).
-      int num_elem_for_resize = MAX2(num_elem_to, Matcher::min_vector_size(elem_bt_to));
+      int num_elem_for_resize = MAX2(num_elem_to, Matcher::min_vector_size(elem_bt_from));
```

Comment:
Looks like a bug to me. 
Can you elaborate, please, how it manifests and does it cause any misbehavior? 
I'm concerned that it should be fixed separately upstream.

Reply:
Yes, we think it is a bug.  I reported a bug : https://bugs.openjdk.java.net/browse/JDK-8258989